README.md

```diff
@@ -20,16 +22,88 @@ Or install it yourself as:
 
     $ gem install pmc_miller
 
+
 ## Usage
 
-TODO: Write usage instructions here
+### Reader
+
+The `PmcMiller::Reader` class can be used to accumulate service data for a
+given key.  This provides a ruby object based approach to
+[grepping metrics](https://github.com/puppetlabs/puppetlabs-puppet_metrics_collector#grepping-metrics).
+
+The following code snippet initializes the reader object to a specified
+`puppet-metrics-collector` path and then accumulates the data for `queue_depth`
+associated with the `puppetdb` service.
+
+```ruby
+require "pmc_miller/reader"
+
+reader = PmcMiller::Reader.new("/path/to/puppet-metrics-collector")
+reader.service = :puppetdb
+data = reader.read(:queue_depth)
+```
+
+The data object returned is an array of PmcMiller::DataPoint objects.  These
+objects are a struct consisting of the `Time` and value of the data point for
+the key that was used for the lookup.  From our previous example, the Data
+object would appear as follows.
+
+```ruby
+puts data    # produces:
+
+#<struct PmcMiller::DataPoint time_string="1970-01-01T00:00:01Z", value=0>
+#<struct PmcMiller::DataPoint time_string="1971-01-01T00:00:01Z", value=10>
+#<struct PmcMiller::DataPoint time_string="1972-01-01T00:00:01Z", value=20>
+```
+
+
+### Analyzer
+PmcMiller contains analyzer classes.  These consume a Data object and return an
+object that contains the following methods.
+
+summary
+: String of the summary of the analysis (pass/fail).
+
+results
+: Hash of calculations created to feed the analysis.
+
+settings
+: Hash of configuration settings used to influence the analysis.
+
+to_json
+: JSON string of the previous three methods.
+
+
+```ruby
+require "pmc_miller/reader"
+
+reader = PmcMiller::Reader.new("/path/to/puppet-metrics-collector")
+reader.service = :puppetdb
+
+data = reader.read(:queue_depth)
+
+qd = PmcMiller::PuppetDB::QueueDepth.new(data)
+
+json = qd.to_json
+
+puts json    #=> {"results":{"rate_of_change":3153600.0},"settings":{},"summary":"fail"}
+```
+
 
 ## Development
 
-After checking out the repo, run `bin/setup` to install dependencies. Then, run `rake spec` to run the tests. You can also run `bin/console` for an interactive prompt that will allow you to experiment.
+After checking out the repo, run `bin/setup` to install dependencies. Then, run
```

Comment:
I noticed that the `bin/setup` script is not included in the PR, but:
1) It would be helpful if it included comments
2) I always prefer to do `bundle install --path vendor/bundle`. Is this possible / preferable?

Reply:
The `--path` flag will be removed from bundler in v3 and as of v2.1 emits deprecation warnings indicating that it should not be used. See the [bundler docs](https://github.com/rubygems/bundler/blob/d4993be66fa2e76b3ca00ea56a51ecab5478b726/UPGRADING.md#bundler-3) for more info.